aws_iam_policies/aws_iam_role_lambda_add_permission.py

```diff
@@ -0,0 +1,66 @@
+import json
+from panther_oss_helpers import resource_lookup
+
+# This is a list of the account numbers included in the organization
+# Example:
+# accounts = [
+#   '123456789012',
+#   '123456789013'
+# ]
+accounts = ['123456789012']
+
+# this is the specific permission that the policy checks
+permission_check = 'lambda:AddPermission'
+
+
+def check_account(resource):
+    content_assumerole = resource['AssumeRolePolicyDocument']
+    principal = content_assumerole['Statement'][0]['Principal']
+
+    if 'AWS' in principal.keys():
+        if type(principal['AWS']) is list:
+            for principal_aws in principal['AWS']:
+                if principal_aws.split(':')[4] not in accounts:
+                    return False
+        else:
+            if principal['AWS'].split(':')[4] not in accounts:
+                return False
+
+    return True
+
+
+def policy(resource):
+    content_inline = resource.get('InlinePolicies', {})
+
+    if content_inline:
+        for policy in content_inline:
+            policy_text = json.loads(content_inline[policy])
+            permissions = policy_text['Statement'][0]['Action']
+
+            if permission_check in permissions:
+                return check_account(resource)
```

Comment:
A couple suggestions:

- We can just loop directly over the `InlinePolicies` dictionary. If it's empty, nothing will happen which is fine.
- We don't actually care about the policy name (`policy` in the above code), just the `policy_text` variable. We can get to this directly by looping over the `.values()` off `InlinePolicies` instead of over the keys
- We should use safe `.get()` lookups
- We should iterate over each statement and check the actions
- Since we only use `permissions` once, we don't need to put it in a variable if it doesn't make things easier to read (this one is more subjective/opinion based suggestion)

Bringing these suggestions all together would look something like this:

```
    for policy_text in (resource.get('InlinePolicies') or {}).values():
        for statement in policy_text.get('Statement', []):
            if permission_check in statement.get('Action', []) and not check_account(resource):
                return False
```

You'll notice in the top level we do `resource.get('InlinePolicies') or {}` instead of using a default value for `.get()`. That's because in Panther top level keys are always present, but may have a `None` value. For many cases that is nice, but breaks down here because python can't iterate `None` and raises an exception. This weird quirk is probably what you ran into when you added the `if content_inline` part.

Reply:
Looping over the `InlinePolicies` dictionary doesn't work if `"InlinePolicies": null`. Same for `"ManagedPolicyNames": null`. Your code above causes a `AttributeError: 'NoneType' object has no attribute 'values'` in those cases.

You need to do a `json.loads` on the policy text because the policy statements are actually text strings.

Reply:
Added the `json.loads()` here.  This should be fine as far as looping over `InlinePolicies` goes, because I loop over `resource.get('InlinePolicies') or {}`. That means if `InlinePolicies` is `null`, we will loop over `{}` instead (an empty dictionary).

```
    for policy_text_str in (resource.get('InlinePolicies') or {}).values():
        policy_text = json.loads(policy_text_str)
        for statement in policy_text.get('Statement', []):
            if permission_check in statement.get('Action', []) and not check_account(resource):
                return False
```